india_compliance/gst_india/report/gstr_1/gstr_1.py

```diff
@@ -933,6 +912,133 @@ def get_columns(self):
         self.columns = self.invoice_columns + self.tax_columns + self.other_columns
 
 
+class GSTR11A11BData:
+    def __init__(self, filters, gst_accounts):
+        self.filters = filters
+
+        self.pe = frappe.qb.DocType("Payment Entry")
+        self.pe_ref = frappe.qb.DocType("Payment Entry Reference")
+        self.gl_entry = frappe.qb.DocType("GL Entry")
+        self.gst_accounts = gst_accounts
+
+    def get_data(self):
+        if self.filters.get("type_of_business") == "Advances":
+            records = self.get_11A_data()
+
+        elif self.filters.get("type_of_business") == "Adjustment":
+            records = self.get_11B_data()
+
+        return self.process_data(records, self.filters.get("type_of_business"))
+
+    def get_11A_data(self):
+        return (
+            self.get_query()
+            .select(
+                self.pe.paid_amount.as_("taxable_value"),
+                Sum(
+                    Case()
+                    .when(
+                        self.gl_entry.account != self.gst_accounts.cess_account,
+                        self.gl_entry.credit_in_account_currency,
+                    )
+                    .else_(0)
+                ).as_("tax_amount"),
+                Sum(
+                    Case()
+                    .when(
+                        self.gl_entry.account == self.gst_accounts.cess_account,
+                        self.gl_entry.credit_in_account_currency,
+                    )
+                    .else_(0)
+                ).as_("cess_amount"),
+            )
+            .where(self.gl_entry.credit_in_account_currency > 0)
+            .groupby(
+                self.pe.name,
+            )
+            .run(as_dict=True)
+        )
+
+    def get_11B_data(self):
+        query = (
+            self.get_query()
+            .join(self.pe_ref)
+            .on(self.pe_ref.name == self.gl_entry.voucher_detail_no)
+            .select(
+                self.pe_ref.allocated_amount.as_("taxable_value"),
+                Sum(
+                    Case()
+                    .when(
+                        self.gl_entry.account != self.gst_accounts.cess_account,
+                        self.gl_entry.debit_in_account_currency,
+                    )
+                    .else_(0)
+                ).as_("tax_amount"),
+                Sum(
+                    Case()
+                    .when(
+                        self.gl_entry.account == self.gst_accounts.cess_account,
+                        self.gl_entry.debit_in_account_currency,
```

Comment:
We can try this implementation:
https://github.com/frappe/erpnext/blob/c5e4e0174782222dc4bc26e3f03f5724d16c99ce/erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py#L674